packages/account/src/Sections/Security/TwoFactorAuthentication/__tests__/two-factor-authentication.spec.tsx

```diff
@@ -0,0 +1,243 @@
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const setup_title = screen.getByText('How to set up 2FA for your Deriv account');
+            expect(setup_title).toBeInTheDocument();
+        });
+    });
+
+    it('should render timeline_1 component title when has_enabled_two_fa is false', async () => {
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const timeline_title_1 = screen.getByText(/Scan the QR code below with your 2FA app. We recommend./i);
+            const authy_link = screen.getByRole('link', { name: 'Authy' });
+            const google_authenticator_link = screen.getByRole('link', { name: 'Google Authenticator' });
+
+            expect(timeline_title_1).toBeInTheDocument();
+            expect(authy_link).toHaveAttribute('href', 'https://authy.com/');
+            expect(google_authenticator_link).toHaveAttribute(
+                'href',
+                'https://github.com/google/google-authenticator/wiki#implementations'
+            );
+        });
+    });
+
+    it('should render QR code when has_enabled_two_fa is false', async () => {
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const qr_code = screen.getByText('QRCode');
+            expect(qr_code).toBeInTheDocument();
+        });
+    });
+
+    it('should render clipboard component to setup 2FA', async () => {
+        WS.authorized.accountSecurity.mockResolvedValueOnce({
+            account_security: {
+                totp: {
+                    secret_key: 'hello123',
+                },
+            },
+        });
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const helper_text = screen.getByText(
+                /If you are unable to scan the QR code, you can manually enter this code instead:/i
+            );
+            const secret_text = screen.getByText('hello123');
+            const clipboard_component = screen.getByTestId('2fa_clipboard');
+
+            expect(helper_text).toBeInTheDocument();
+            expect(secret_text).toBeInTheDocument();
+            expect(clipboard_component).toBeInTheDocument();
+        });
+    });
+
+    it('should render step-2 title for setting up 2FA', async () => {
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const step_2_title = screen.getByText(/Enter the authentication code generated by your 2FA app:/i);
+            expect(step_2_title).toBeInTheDocument();
+        });
+    });
+
+    it('should render digitform component when 2FA is disabled', async () => {
+        renderComponent({ store_config: store });
+
+        await waitFor(() => {
+            const digitform = screen.getByTestId('digitform_2fa_disabled');
+            expect(digitform).toBeInTheDocument();
+        });
+    });
```

Comment:
why do we need async await in all these test cases? could you please check this

Reply:
Inside 2FA component everytime we render , useEffect runs 2 functions - generateQrCode and getDigitStatus which update states internally. We need waitFor() so that state updates showup otherwise testcases would fail. Additionally, since all this is async in nature I added async as the functions make api calls and state updates happen asynchronously.